pkg/virt-launcher/virtwrap/device/hostdevice/sriov/hostdev.go

```diff
@@ -39,8 +39,16 @@ import (
 )
 
 func CreateHostDevices(vmi *v1.VirtualMachineInstance) ([]api.HostDevice, error) {
-	SRIOVInterfaces := vmispec.FilterSRIOVInterfaces(vmi.Spec.Domain.Devices.Interfaces)
+	SRIOVInterfaces := vmispec.FilterInterfacesSpec(vmi.Spec.Domain.Devices.Interfaces, func(iface v1.Interface) bool {
+		if iface.SRIOV == nil {
+			return false
+		}
+		ifaceStatus := vmispec.LookupInterfaceStatusByName(vmi.Status.Interfaces, iface.Name)
+		return ifaceStatus != nil && vmispec.ContainsInfoSource(ifaceStatus.InfoSource, vmispec.InfoSourceMultusStatus)
```

Comment:
Up until now we kept virt-launcher from knowing those details (look in vmi status, compare it with spec..) and only compare between the domain and desired vmi spec.
Similar check already done at virt-handler https://github.com/kubevirt/kubevirt/blob/0379bbe4fd7e10fc6f9d1a96abe81e7bffafe5f2/pkg/virt-handler/vm.go#L3156-L3159
Note that there is also ratelimiting on SR-IOV hotplug operation.

Can we avoid doing this check in virt-launcher as well?

Reply:
I dont have better solution at the moment, once I have I will post a follow up PR.